crypto/BRCryptoNetwork.c

```diff
@@ -203,10 +204,13 @@ cryptoNetworkCreateAsETH (const char *uids,
 private_extern BRCryptoNetwork
 cryptoNetworkCreateAsGEN (const char *uids,
                           const char *name,
+                          BRCryptoCurrency currency,
                           uint8_t isMainnet) {
     BRCryptoNetwork network = cryptoNetworkCreate (uids, name);
     network->type = BLOCK_CHAIN_TYPE_GEN;
     network->u.gen.mainnet = isMainnet;
+    network->u.gen.net = gwmNetworkCreate(cryptoCurrencyGetCode(currency));
+    network->currency = cryptoCurrencyTake(currency);
```

Comment:
Not sure I need to keep the currency since it was only needed to create the BRGenericNetwork

Reply:
I don't think you need it; doesn't hurt to hold it (as long as the 'cryptoNetworkRelease()` call `cryptoCurrencyGive(...)`